### PR TITLE
GitHub Docs Update

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Updated/reorganized some of the GitHub-specific docs.

**What changed?**

- Moved old PR template to Release Process notes
- Created new PR template pre-tagged with `@edx/masters-devs-gta`
- Added `CODEOWNERS` file to auto-assign team as reviewers
- Moved GitHub files to `.github/` folder

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [ ] ~Translations up to date~
- [ ] ~JS minified, SASS compiled~
- [ ] ~Test suites passing on Jenkins~
- [ ] ~Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)~

JIRA: n/a
FIY: @edx/masters-devs-gta

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+**Developer Checklist**
+- [ ] Reviewed the [release process](./release_process.md)
+- [ ] Translations up to date
+- [ ] JS minified, SASS compiled
+- [ ] Test suites passing on Jenkins
+- [ ] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)
+
+JIRA: [EDUCATOR-XXXX](https://openedx.atlassian.net/browse/EDUCATOR-XXXX)
+
+FIY: @edx/masters-devs-gta
```
Comment (Andytr1) [inline .github/pull_request_template.md:18]: I admit to forgetting/omitting this tag on occasion. However, I would not check it in because teams/people change.
Same can be said for 'EDUCATOR-' in Jira labels. Different teams use different prefixes. Perhaps just change it to JIRA:(https://openedx.atlassian.net/browse/)

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Code owners for edx-ora2, open response assessment repo
+
+# These owners will be the default owners for everything in
+# the repo. Unless a later match takes precedence, they will
+# be requested for review when someone opens a pull request.
+*       @edx/masters-devs-gta @mattcarter
```
Comment (Andytr1) [inline .github/CODEOWNERS:6]: not sure about calling out an individual vs a team - i.e: remove `@mattcarter` ?

Comment (nsprenkle) [reply .github/CODEOWNERS:6]: I would like to just do a team, but right now the `edx/masters-devs-gta` can't be called as a reviewer... gotta figure that out.

File: .github/pull_request_template.md (hunk 1 above)
Comment (nsprenkle) [reply .github/pull_request_template.md:18]: My reasoning was to eliminate as much repetitious work as possible. Since, for the immediate future, we own the code, I think it's saving us having to type it. Could be changed when the code owners change

Comment (nsprenkle) [reply .github/pull_request_template.md:18]: I think it might be valid to remove the `EDUCATOR-XXXX` blob though if other folks (e.g. ARCH) touch this

File: .github/CODEOWNERS (hunk 2 above)
Comment (mattcarter) [reply .github/CODEOWNERS:6]: We should find alignment with `openedx.yaml`; between `supporting_teams` and `owner`  which is us and me, respectively ATM. I think the setup you currently have proposed is fine

**Comment by nsprenkle** [comment]
I've come around and agree w/ @Andytr1's suggestions. Updated docs.

**Comment by nsprenkle** [comment]
jenkins run pr

File: .github/pull_request_template.md (hunk 1 above)
Comment (mattcarter) [reply .github/pull_request_template.md]: Agreed.  Having a ticket skeleton is a Good Idea. I'm of the opinion our fellow committers are  capable in detecting they'll need to use the project prefix for their ticket and not just the ticket number.

And @Andytr1 as we own this repo, we can make updating the template part of our eventual off-boarding of ownership.  Aka, this makes un-owning it an overt act.

File: .github/release_process.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Change Review and Release Process
+
+Before Merging a pull request:
+
+- [ ] Bump version number in [setup.py](../setup.py) and [package.json](../package.json)
+- [ ] Run `make javascript sass` if your changes include JS/CSS changes
+- [ ] Get a green Travis build for this PR
```
Comment (Andytr1) [inline .github/release_process.md:7]: what about adding  a step to: `make check_translations_up_to_date` and check in any files generated by that command. To remind us of the command to update translations?

Comment (nsprenkle) [reply .github/release_process.md:7]: That's in the developer checklist right now in `pull_request_template`. I'm still unsure of which pieces I want to put where but everything should still be covered.

Comment (nsprenkle) [reply .github/release_process.md:7]: @Andytr1, I updated the release process to have that script
